sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/json/JsonSuite.scala

```diff
@@ -2127,4 +2127,39 @@ class JsonSuite extends QueryTest with SharedSQLContext with TestJsonData {
       assert(df.schema === expectedSchema)
     }
   }
+
+  test("schema inferring touches less data if samplingRation < 1.0") {
```

Comment:
Hi @MaxGekk , since these test-cases in `JsonSuite` are not related to the changes made in this patch, can it be raised as a separate PR?

Reply:
@sujithjay I created separate PR for the tests: https://github.com/apache/spark/pull/20963